### ISSUE TITLE
Exception when using ShouldBeEquivalentTo on lists of objects with Guid property

### ISSUE THREAD

**Issue by yinnonsanders** [fluentassertions/fluentassertions#613]
Here's an example of code that causes the exception:

        class Dto
        {
            public Guid Id { get; set; }
            public string Name { get; set; }
        }

        [TestMethod]
        public void Test()
        {
            var list1 =
                new List<Dto>
                {
                    new Dto { Id = Guid.NewGuid(), Name = "Name" }
                };
            var list2 =
                new List<Dto>
                {
                    new Dto { Id = Guid.NewGuid(), Name = "Name" }
                };
            list1.ShouldBeEquivalentTo(list2);
        }

This results in a System.FormatException.
In this case, the assertion should fail, but there is also a case in which the assertion should succeed that throws the same exception:

        [TestMethod]
        public void Test()
        {
            var list1 =
                new List<Dto>
                {
                    new Dto { Id = Guid.NewGuid(), Name = "Name" }
                };
            var list2 =
                new List<Dto>
                {
                    new Dto { Id = Guid.NewGuid(), Name = "Name" }
                };
            var nestedList1 = new List<List<Dto>> { list1, list2 };
            var nestedList2 = new List<List<Dto>> { list2, list1 };
            nestedList1.ShouldBeEquivalentTo(nestedList2);
        }

**Comment by jnyrup** [comment]
When using the develop branch:

The first test correctly throws with
```
Result Message:	
Expected item[0].Id to be {b477ff95-efe4-4d50-9c98-87276b754500}, but found {485d2674-ae6b-4de0-8a70-76170d43dee5}.

With configuration:
- Use declared types and members
- Compare enums by value
- Match member by name (or throw)
- Be strict about the order of items in byte arrays
```

The second test succeeds as expected.

**Comment by dennisdoomen** [comment]
I wonder whether this is fixed by https://github.com/fluentassertions/fluentassertions/releases/tag/4.19.3

**Comment by jnyrup** [comment]
in 4.19.3 it throws:
```
Result StackTrace:	
at System.Text.StringBuilder.AppendFormatHelper(IFormatProvider provider, String format, ParamsArray args)
   at System.String.FormatHelper(IFormatProvider provider, String format, ParamsArray args)
   at System.String.Format(String format, Object[] args)
   at FluentAssertions.Execution.MessageBuilder.FormatArgumentPlaceholders(String failureMessage, Object[] failureArgs) in C:\projects\fluentassertions-vf06b\Src\Core\Execution\MessageBuilder.cs:line 66
   at FluentAssertions.Execution.MessageBuilder.Build(String message, Object[] messageArgs, String reason, ContextDataItems contextData) in C:\projects\fluentassertions-vf06b\Src\Core\Execution\MessageBuilder.cs:line 39
   at FluentAssertions.Execution.AssertionScope.FailWith(String message, Object[] args) in C:\projects\fluentassertions-vf06b\Src\Core\Execution\AssertionScope.cs:line 190
   at FluentAssertions.Equivalency.EnumerableEquivalencyValidator.LooselyMatchAgainst[T](IList`1 subjects, Object expectation, Int32 expectationIndex) in C:\projects\fluentassertions-vf06b\Src\Core\Equivalency\EnumerableEquivalencyValidator.cs:line 123
   at FluentAssertions.Equivalency.EnumerableEquivalencyValidator.AssertElementGraphEquivalency[T](T[] subjects, Object[] expectations) in C:\projects\fluentassertions-vf06b\Src\Core\Equivalency\EnumerableEquivalencyValidator.cs:line 79
   at FluentAssertions.Equivalency.EnumerableEquivalencyValidator.Execute[T](T[] subject, Object[] expectation) in C:\projects\fluentassertions-vf06b\Src\Core\Equivalency\EnumerableEquivalencyValidator.cs:line 39
   at lambda_method(Closure )
--- End of inner exception stack trace ---
    at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at System.Delegate.DynamicInvokeImpl(Object[] args)
   at FluentAssertions.Equivalency.GenericEnumerableEquivalencyStep.Handle(IEquivalencyValidationContext context, IEquivalencyValidator parent, IEquivalencyAssertionOptions config) in C:\projects\fluentassertions-vf06b\Src\Core\Equivalency\GenericEnumerableEquivalencyStep.cs:line 61
   at FluentAssertions.Equivalency.EquivalencyValidator.AssertEqualityUsing(IEquivalencyValidationContext context) in C:\projects\fluentassertions-vf06b\Src\Core\Equivalency\EquivalencyValidator.cs:line 63
   at FluentAssertions.Equivalency.EquivalencyValidator.AssertEquality(EquivalencyValidationContext context) in C:\projects\fluentassertions-vf06b\Src\Core\Equivalency\EquivalencyValidator.cs:line 35
   at FluentAssertions.AssertionExtensions.ShouldBeEquivalentTo[T](T subject, Object expectation, Func`2 config, String because, Object[] becauseArgs) in C:\projects\fluentassertions-vf06b\Src\Shared\AssertionExtensions.cs:line 744
   at FluentAssertions.AssertionExtensions.ShouldBeEquivalentTo[T](T subject, Object expectation, String because, Object[] becauseArgs) in C:\projects\fluentassertions-vf06b\Src\Shared\AssertionExtensions.cs:line 703
   at UnitTests2.AssertTests.Test613a() in C:\Users\jonas\Documents\Visual Studio 2015\Projects\ConsoleApplication2\UnitTests2\AssertTests.cs:line 621
Result Message:	
Test method UnitTests2.AssertTests.Test613 threw exception: 
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.FormatException: Index (zero based) must be greater than or equal to zero and less than the size of the argument list.
```

`FormatArgumentPlaceholders(string failureMessage, object[] failureArgs)` is called with
`Expected item[0].Id to be {196f8c8a-cd48-42c4-99e0-26d8097d26a1}, but found {b3b0b7de-21c3-4176-ae7c-84b74075ce06}.` and the empty array.
So it interprets the first guid as a placeholder, probably because the `ToString` of `GUID` is `{value-of-guid}`.

**Comment by dennisdoomen** [comment]
The main problem is that the `EnumerableEquivalencyValidator` is trying to format something that was already formatted. I think we need an overload on `AssertionScope` that just adds the preformatted message to the internal list it keeps.

**Comment by dennisdoomen** [comment]
Should be fixed on `master` though.

**Comment by jnyrup** [comment]
@dennisdoomen don't you mean `develop`?
The only changes between master and 4.19.3 are documentation?

**Comment by dennisdoomen** [comment]
Yeah, but a bug like that can't wait for v5.

**Comment by jnyrup** [comment]
Sorry, misread your "should be fixed" as "should've been fixed".

**Comment by dennisdoomen** [comment]
Fixed in 4.19.4
